docs/field-level-encryption.md

```diff
@@ -243,3 +149,97 @@ await connection.openUri(`mongodb://localhost:27017`, {
+This is MongoDB's official package for setting up encryption keys.
+
+```sh
+npm install mongodb-client-encryption
+```
+
+You also need to make sure you've installed [mongocryptd](https://www.mongodb.com/docs/manual/core/queryable-encryption/reference/mongocryptd/).
+mongocryptd is a separate process from the MongoDB server that you need to run to work with field level encryption.
+You can either run mongocryptd yourself, or make sure it is on the system PATH and the MongoDB Node.js driver will run it for you.
+[You can read more about mongocryptd here](https://www.mongodb.com/docs/v5.0/reference/security-client-side-encryption-appendix/#mongocryptd).
+
+Once you've set up and run mongocryptd, first you need to create a new encryption key as follows.
+Keep in mind that the following example is a simple example to help you get started.
+The encryption key in the following example is insecure; MongoDB recommends using a [KMS](https://www.mongodb.com/docs/v5.0/core/security-client-side-encryption-key-management/).
+
+```javascript
+const { ClientEncryption } = require('mongodb');
+const mongoose = require('mongoose');
+
+run().catch(err => console.log(err));
+
+async function run() {
+  /* Step 1: Connect to MongoDB and insert a key */
+
+  // Create a very basic key. You're responsible for making
+  // your key secure, don't use this in prod :)
+  const arr = [];
+  for (let i = 0; i < 96; ++i) {
+    arr.push(i);
+  }
+  const key = Buffer.from(arr);
+
+  const keyVaultNamespace = 'client.encryption';
+  const kmsProviders = { local: { key } };
+
+  const uri = 'mongodb://127.0.0.1:27017/mongoose_test';
+  const conn = await mongoose.createConnection(uri, {
+    autoEncryption: {
+      keyVaultNamespace,
+      kmsProviders
+    }
+  }).asPromise();
+  const encryption = new ClientEncryption(conn.getClient(), {
+    keyVaultNamespace,
+    kmsProviders,
+  });
+
+  const _key = await encryption.createDataKey('local', {
+    keyAltNames: ['exampleKeyName'],
+  });
+}
+```
+
+Once you have an encryption key, you can create a separate Mongoose connection with a [`schemaMap`](https://mongodb.github.io/node-mongodb-native/5.6/interfaces/AutoEncryptionOptions.html#schemaMap) that defines which fields are encrypted using JSON schema syntax as follows.
+
+```javascript
+/* Step 2: connect using schema map and new key */
+await mongoose.connect('mongodb://127.0.0.1:27017/mongoose_test', {
+  // Configure auto encryption
+  autoEncryption: {
+    keyVaultNamespace,
+    kmsProviders,
+    schemaMap: {
+      'mongoose_test.tests': {
+        bsonType: 'object',
+        encryptMetadata: {
+          keyId: [_key]
+        },
+        properties: {
+          name: {
+            encrypt: {
+              bsonType: 'string',
+              algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic'
+            }
+          }
+        }
+      }
+    }
```

Comment:
Isnt this auto-generated by the connection `_buildSchemaMap` / `_buildEncryptedFields`?

Reply:
Yeah.  I left this section untouched from the old Mongoose FLE docs because this pattern still works.

For clarity, I updated the header now to make it more obvious that this documentation is for manual configuration of FLE instead of automatic configuration.